packages/rule-https-only/src/rule.ts

```diff
@@ -0,0 +1,69 @@
+/**
+ * @fileoverview sonarwhal rule that checks if your site is using HTTPS and if it has mixed content.
+ */
+
+import { Category } from 'sonarwhal/dist/src/lib/enums/category';
+import { RuleContext } from 'sonarwhal/dist/src/lib/rule-context';
+
+import { IRule, RuleMetadata, Event } from 'sonarwhal/dist/src/lib/types';
+import { debug as d } from 'sonarwhal/dist/src/lib/utils/debug';
+import { RuleScope } from 'sonarwhal/dist/src/lib/enums/rulescope';
+import { isHTTPS } from 'sonarwhal/dist/src/lib/utils/misc';
+
+const debug: debug.IDebugger = d(__filename);
+
+/*
+ * ------------------------------------------------------------------------------
+ * Public
+ * ------------------------------------------------------------------------------
+ */
+
+export default class HttpsOnlyRule implements IRule {
+    private targetIsServedOverHTTPS: boolean = false;
+
+    public static readonly meta: RuleMetadata = {
+        docs: {
+            category: Category.security,
+            description: `sonarwhal rule that checks if your site is using HTTPS and if it has mixed content.`
+        },
+        id: 'https-only',
+        schema: [],
+        scope: RuleScope.site
+    }
+
+    public constructor(context: RuleContext) {
+
+        const validateTarget = async (event: Event): Promise<void> => {
+            const { resource } = event;
+
+            if (!isHTTPS(resource)) {
+                debug('HTTPS no detected');
+
+                await context.report(resource, null, 'The site should be HTTPS');
+
+                return;
+            }
+
+            debug('HTTPS detected');
+
+            this.targetIsServedOverHTTPS = true;
+
+            return;
+        };
+
+        const validate = async (event: Event) => {
+            if (!this.targetIsServedOverHTTPS) {
+                return;
+            }
+
+            const { resource } = event;
+
+            if (!isHTTPS(resource)) {
+                await context.report(resource, null, 'Should be served over HTTPS');
+            }
+        };
+
+        context.on('fetch::start::target', validateTarget);
+        context.on('fetch::start', validate);
```

Comment:
Shouldn't the manifest be included?

Reply:
Also don't we need to subscribe using fetch::start::* ?

Reply:
we only have `fetch::start`, `fetch::start::target` and `fetch::start::manifest`, `fetch::start::*` doesn't work for `fetch::start`

Reply:
ok, in any case we should be changing to the `fetch::end` so does this matter?